tap_urban_airship/__init__.py

```diff
@@ -86,7 +86,9 @@ def sync_entity(entity, primary_keys, date_keys=None, transform=None):
 
         row = transform_row(row, schema)
         if date_keys:
-            last_touched = max(row[date_key] for date_key in date_keys)
+            if not any(date_key in row for date_key in date_keys):
```

Comment:
At first glance, it seems like if we are getting key errors here then we need to adjust how we call this function.

If I'm wrong here, could you add some comments to clarify your logic here?

Reply:
Ideally, this error will never be raised. But at the moment we are having an issue caused by missing `last_updated` key in the row. So I added a `KeyError` raising here in order to, if suddenly there is a second key missing, we noticed it instead of unclear `ValueError: max() arg is an empty sequence`.

Reply:
Ah that makes sense.

I think the various list comprehensions here are complicated enough that it would warrant a comment. Probably right above line 88.

Something along the lines of:

```
# Rows can have various values for various sets of keys (See the calls to
# `sync_entitity` in `do_sync`).
#
# To handle this we:
#
# 1. Get _all_ the values for all the keys that are actually present in
# the row (not every row has every key).
#
# 2. Take the max of those values as the bookmark for that entity.
#
# A KeyError is raised in the event that a row has none of the bookmark
# keys.
```

Reply:
Done.